tensorflow_model_optimization/python/core/sparsity_tf2/lthpruner_test.py

```diff
@@ -0,0 +1,486 @@
+    self.assertAllEqual(np.count_nonzero(mask_after_pruning), 97)
+
+  def testReloadWeightsatInitialization(self):
+    weight1 = tf.Variable(np.linspace(1.0, 100.0, 100), name="weights")
+
+    save_round = 0
+    n_rounds = 20
+    end_iter = 100
+    frequency, prune_ratio_per_round = get_lth_sparsity(save_round, n_rounds, self.target_sparsity, end_iter)
+    pruning_schedule = make_pruning_schedule(1 - prune_ratio_per_round, save_round, end_iter, frequency)
+    
+    p = pruner.LTHPruner(
+        pruning_schedule=pruning_schedule,
+        save_iteration=self.save_init,
+        block_size=self.block_size,
+        block_pooling_type=self.block_pooling_type)
+
+    optimizer = self.dummy_optimizer
+    optimizer.iterations.assign(0)
+    expected_saved_initialization = None
+
+    def _train(weight):
+      expected_saved_initialization = tf.ones_like(weight, dtype=weight.dtype.base_dtype) * -1
+      p.create_slots(optimizer, weight)
+      for i in tf.range(0, 2):
+        p.preprocess_weights(optimizer, weight, self.grad(weight))
+        if i == save_round:
+          expected_saved_initialization = weight.read_value()
+        weight.assign(tf.math.add(weight, sample_noise(i)))
+        p.postprocess_weights(optimizer, weight, self.grad(weight))
+        optimizer.iterations.assign_add(1) # save weights right before iteration
+      return expected_saved_initialization
+
+    expected_saved_initialization = _train(weight1)
+    initialization_slot = optimizer.get_slot(weight1, "original_initialization")
+    self.assertAllEqual(initialization_slot, expected_saved_initialization)
+
+    mask_after_pruning = optimizer.get_slot(weight1, "mask").read_value()
+    masked_weight_expected = tf.math.multiply(initialization_slot, mask_after_pruning)
+    self.assertAllEqual(np.count_nonzero(masked_weight_expected), 97)
+    self.assertAllEqual(np.count_nonzero(mask_after_pruning), 97)
+
+    weight = tf.Variable(np.linspace(1.0, 100.0, 100), name="weights")
+    optimizer.iterations.assign(0)
+    expected_saved_initialization = tf.function(_train)(weight)
+    initialization_slot = optimizer.get_slot(weight, "original_initialization")
+    self.assertAllEqual(initialization_slot, expected_saved_initialization)
+
+    mask_after_pruning = optimizer.get_slot(weight, "mask").read_value()
+    masked_weight_expected = tf.math.multiply(initialization_slot, mask_after_pruning)
+    self.assertAllEqual(np.count_nonzero(masked_weight_expected), 97)
+    self.assertAllEqual(np.count_nonzero(mask_after_pruning), 97)
+
+  
+  def testReloadWeightsIterationK(self):
+    weight = tf.Variable(np.linspace(1.0, 100.0, 100), name="weights")
+
+    save_round = 5
+    n_rounds = 24
+    end_iter = 100
+    frequency, prune_ratio_per_round = get_lth_sparsity(save_round, n_rounds, self.target_sparsity, end_iter)
+    pruning_schedule = make_pruning_schedule(1 - prune_ratio_per_round, save_round + 1, end_iter, frequency)
```

Comment:
Let's add a comment explicitly specifying what step # this will prune & reload at. I'm guessing it's the second arg, but it's not immediately obvious from this pruning schedule definition.